lib/puppet/provider/uri_parsing.rb

```diff
@@ -16,8 +16,8 @@
 
 require File.expand_path(File.join(File.dirname(__FILE__), '../provider', 'login'))
 
-def uri_validation(data)
-  @client = OneviewSDK::Client.new(login)
+def uri_validation(data, client)
+  @client = client.nil? ? OneviewSDK::Client.new(login) : client
```

Comment:
What is the use case for this change?
Also, could be written as
`@client = client || OneviewSDK::Client.new(login)`